source/ast/symbols/ClassSymbols.cpp

```diff
@@ -699,18 +703,45 @@ void ClassType::handleImplements(const ImplementsClauseSyntax& implementsClause,
                                                           /* allowDerivedReturn */ false);
             }
 
-            findIfaces(*iface, ifaces, seenIfaces);
+            findIfaces(*iface, ifaces, seenIfaces, true);
+            findIfaces(*iface, ifacesImmediate, seenIfacesImmediate, false);
         }
     }
 
     implementsIfaces = ifaces.copy(comp);
+    implementsIfacesImmediate = ifacesImmediate.copy(comp);
 }
 
 void ClassType::serializeTo(ASTSerializer& serializer) const {
     if (firstForward)
         serializer.write("forward", *firstForward);
     if (genericClass)
         serializer.writeLink("genericClass", *genericClass);
+
+    serializer.writeProperty("extends");
+    serializer.startObject();
+    if (getBaseClass()) {
+        serializer.write("baseClassName", getBaseClass()->getCanonicalType().toString());
```

Comment:
You should just write the type instead of diving down and toString()ing it yourself.